tests/test_ell1h.py

```diff
@@ -12,111 +13,239 @@
+            sigma,
+        )
+
+
+def test_no_H3_H4(toasJ0613, simple_par_str):
+    """ Test no H3 and H4 in model.
+    """
+    no_H3_H4 = simple_par.replace("H4 2.0262048E-7  1       1.1276173E-7", "")
+    no_H3_H4 = no_H3_H4.replace("H3 2.7507208E-7  1       1.5114416E-7", "")
+    print(no_H3_H4)
+    no_H3_H4_model = model.get_model(StringIO(no_H3_H4))
+    assert no_H3_H4_model.H3.value == None
+    assert no_H3_H4_model.H4.value == None
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, no_H3_H4_model)
+    f.fit_toas()
+
 
-    def test_J0613_H4(self):
-        log = logging.getLogger("TestJ0613.fit_tests")
-        f = ff.GLSFitter(self.toasJ0613, self.modelJ0613)
+def test_H3_H4_pairs(toasJ0613, simple_par_str):
+    """ Testing if the different H3, H4 combination breaks the fitting. the
+        fitting result will not be checked here.
+    """
+    simple_model = model.get_model(StringIO(simple_par_str))
+
+    test_toas = toasJ0613[::20]
+    f = ff.WLSFitter(test_toas, simple_model)
+    f.fit_toas()
+
+
+def zero_H4(toasJ0613, simple_par_str):
+    H4_zero = simple_par_str.replace(
+        "H4 2.0262048E-7  1       1.1276173E-7", "H4 0  1  0"
+    )
+    H4_zero_model = model.get_model(StringIO(H4_zero))
+    assert H4_zero_model.H4.value == 0.0
+    assert H4_zero_model.H3.value != 0.0
+    f = ff.WLSFitter(test_toas, H4_zero_model)
+    f.fit_toas()
+
+
+def zero_H3(toasJ0613, simple_par_str):
+    H3_zero = simple_par_str.replace(
+        "H3 2.7507208E-7  1       1.5114416E-7", "H3 0  1  0"
+    )
+    H3_zero_model = model.get_model(StringIO(H3_zero))
+    assert H3_zero_model.H3.value == 0.0
+    assert H3_zero_model.H4.value != 0.0
+    f = ff.WLSFitter(test_toas, H3_zero_model)
+    with pytest.raises(ValueError):
         f.fit_toas()
-        f.model.free_params = ("H3", "H4")
-        for pn, p in f.model.get_params_dict("free", "quantity").items():
-            op = getattr(f.model_init, pn)
-            diff = np.abs(p.value - op.value)
-            sigma = diff / op.uncertainty_value
-            # Fit th
-            assert (
-                sigma < 0.7
-            ), "refit %s is %lf sigma different from original value" % (pn, sigma)
-
-    def test_J0613_STIG(self):
-        log = logging.getLogger("TestJ0613.fit_tests_stig")
-        f = ff.GLSFitter(self.toasJ0613, self.modelJ0613_STIG)
+
+
+def zero_H3_H4_fit_H3_H4(toasJ0613, simple_par_str):
+    H4_zero = simple_par_str.replace(
+        "H4 2.0262048E-7  1       1.1276173E-7", "H4 0  1  0"
+    )
+    H3H4_zero = H4_zero.replace("H3 2.7507208E-7  1       1.5114416E-7", "H3 0  1  0")
+    H3H4_zero_model = model.get_model(StringIO(H3H4_zero))
+    assert H3H4_zero_model.H3.value == 0.0
+    assert "H3" in H3H4_zero_model.free_params
+    assert H3H4_zero_model.H4.value == 0.0
+    assert "H4" in H3H4_zero_model.free_params
+    f = ff.WLSFitter(test_toas, H3H4_zero_model)
+    # Fitting H4 with H3 == 0, will give a ValueError
```

Comment:
Wait, shouldn't this work? Like, shouldn't  one be able to start with H3==H4==0 and fit for both? Presumably the fit value for H3 will be nonzero

Reply:
Yeah, I agree with @aarchiba here.  We should be able to start a fit with H3 == H4 == 0 and get a reasonable result for a system where there is a small amount of Shapiro signal.  Users will *definitely* try this.

Reply:
It will probably require special-casing the derivatives, maybe just returning zero for derivatives with respect to H4 if H3 is zero. But that should work (fitting will require more than one step, almost certainly).